src/Units/UnitModelScheme.java

```diff
@@ -0,0 +1,62 @@
+package Units;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import Terrain.TerrainType;
+import Units.UnitModel.UnitRoleEnum;
+
+/**
+ * Provides the primary interface for the game to get info about game-specific units.
+ */
+public abstract class UnitModelScheme implements Serializable
+{
+  private static final long serialVersionUID = 1L;
+  
+  /** Indicates whether, for the given map/game, all the units can be instantiated */
+  public boolean schemeValid = true;
+
+  /**
+   * When the UI wants to show a unit type, it should use this string.
+   * Typically the name of whatever passes for infantry, but can be anything.
+   */
+  public abstract String getIconicUnitName();
+  
+  // Fetches a shopping list and a full unit list for a generic commander.
+  public abstract GameReadyModels getGameReadyModels();
+
+  // Holds the data for a single Commander's unit selection.
+  public static class GameReadyModels
+  {
+    public HashMap<TerrainType, ArrayList<UnitModel>> shoppingList = new HashMap<TerrainType, ArrayList<UnitModel>>();
+    public ArrayList<UnitModel> unitModels = new ArrayList<UnitModel>();
+  }
+  
+  public static UnitModel getModelFromString(String pName, ArrayList<UnitModel> models)
```

Comment:
If we are really going to use Strings for the canonical ID/name of units, we might consider throwing an exception if we fail to match it to a model.
Granted, if we get a `null` model back, it'll probably cause an exception later on, but may as well fail faster.

Reply:
Throwing an exception would make the MapMaster constructor uglier than it needs to be: if getModelFromString fails, you just... don't get the unit.
So long as we let users play with a unit set that doesn't match the map, this isn't an exceptional execution path... and it's not exceptional anyhow, since it's used for checking map/scheme compatibility in the first place.